### PR TITLE
feature/AT-SSO-UI Consume User API

### PR DESCRIPTION
**Pull request by BrendaMorales9719**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,11 +13,13 @@
   "dependencies": {
     "axios": "^0.20.0",
     "bcryptjs": "^2.4.3",
+    "body-parser": "^1.19.0",
     "connect-flash": "^0.1.1",
     "express": "^4.17.1",
     "express-handlebars": "^5.1.0",
     "express-session": "^1.17.1",
     "method-override": "^3.0.0",
+    "mongoose": "^5.10.8",
```
Comment (at-internship) [inline package.json]: I think we don't need this dependency

File: src/controllers/admin.controller.js [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
+    }
+
+    res.render('users/admin', { usersArray });
 };
 
 //Render user form
-adminCtrl.renderUserForm = (req,res) => {
-    res.render('hola aqui va user form');
+adminCtrl.renderUserForm = (req, res) => {
+    res.render('users/createUserForm');
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: why?
Try: `res.render('admin/users/createUserForm');`

File: src/models/User.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+'use strict'
```
Comment (at-internship) [inline src/models/User.js]: we don't need this file

**Comment by BrendaMorales9719** [comment]
image

**Comment by BrendaMorales9719** [comment]
visualizacio-de-datos-userCollection

File: src/services/user.service.js [hunk 4]
```diff
@@ -1,52 +1,41 @@
-const { default: Axios } = require("axios")
-
-const axios = require ("axios");
+const axios = require("axios");
+const User = require("../models/User");
```
Comment (at-internship) [inline src/services/user.service.js]: we don't need this Model

File: src/services/user.service.js [hunk 5]
```diff
@@ -1,52 +1,41 @@
...
 const USERS_SERVICE_URI = "https://at-sso-api.herokuapp.com/api/v1/user";
```
Comment (at-internship) [inline src/services/user.service.js]:
```
const AT_SSO_SERVICE_URI = process.env.AT_SSO_SERVICE_URI;
console.log("AT_SSO_SERVICE_URI:" + AT_SSO_SERVICE_URI);
```

File: src/services/user.service.js [hunk 6]
```diff
@@ -1,52 +1,41 @@
...
-//Metodo 2
-/*const userService = {};
+const userService = {};
 
-userService.getAllUsers = () =>{
-    return axios ({
+userService.getUsers = function () {
+    return new Promise((resolve, reject) => axios({
         method: "GET",
         url: USERS_SERVICE_URI,
```
Comment (at-internship) [inline src/services/user.service.js]: `url: AT_SSO_SERVICE_URI,`

File: src/views/users/createUserForm.hbs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<div class="col-md-4 mx-auto">
```
Comment (at-internship) [inline src/views/users/createUserForm.hbs:1]: Why?
Try this PATH: src/views/admin/users/createUserForm.hbs

**Review by at-internship — CHANGES_REQUESTED** [review]
Please take a look

File: package.json [hunk 8]
```diff
@@ -13,11 +13,13 @@
   "dependencies": {
     "axios": "^0.20.0",
     "bcryptjs": "^2.4.3",
+    "body-parser": "^1.19.0",
```
Comment (pinto1997) [inline package.json]: Please check if you really need this dependency

File: src/controllers/admin.controller.js [hunk 9]
```diff
@@ -1,18 +1,38 @@
...
 }
 
 //Add user
-adminCtrl.addUser = (req,res) => {
-    res.render('hola aqui va add user');
+adminCtrl.addUser = (req, res) => {
+    res.render('Hola aquí va la parte de agregar usuario');
```
Comment (pinto1997) [inline src/controllers/admin.controller.js:35]: Change this to 'add user render'

File: src/server.js [hunk 10]
```diff
@@ -1,21 +1,41 @@
...
 //Static files
 app.use(express.static(path.join(__dirname, 'public')));
 
-module.exports = app;
\ No newline at end of file
+module.exports = app;
+
+
+
+
```
Comment (pinto1997) [inline src/server.js]: Please delete these empty spaces, you just need one of them

File: src/models/User.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    UserSchema.methods.encrypPassword = async pwd => {
+        const salt = await bcrypt.genSalt(10);
+        return await bcrypt.hash(pwd, salt);
+    };
+
+    UserSchema.methods.matchPassword = async function(pwd) {
+        return await bcrypt.compare(pwd, this.pwd)
+    }
+
+    module.exports = model('User', UserSchema);
```
Comment (pinto1997) [inline src/models/User.js]: Add an empty line here

File: src/services/user.service.js [hunk 12]
```diff
@@ -1,52 +1,41 @@
...
+                name: 'String',
+                firstName: 'String',
+                lastName: 'String',
+                email: 'String',
+                password: 'String (hash)',
+                status: 'Integer 0,1',
+            },
+        })
+    }, userService
 }
```
Comment (pinto1997) [inline src/services/user.service.js:43]: Add an empty line here

File: src/views/layouts/main.hbs [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>AT-SSO</title>
+</head>
+<body>
+    {{{body}}}
+</body>
+</html>
```
Comment (pinto1997) [inline src/views/layouts/main.hbs]: Add an empty line here

File: src/views/users/admin.hbs [hunk 14]
```diff
@@ -0,0 +1,6 @@
+<h2>Users</h2>
+<div class="usersList">
+{{#each usersArray}}
+<h5>Nombre: {{name}}​​ - Primer apellido: {{firstname}}​​ - Segundo apellido: {{lastname}}​​ - Correo: {{email}}​​ - Estatus: {{status}}​​</h5>
+{{/each}}​​
+</div>
```
Comment (pinto1997) [inline src/views/users/admin.hbs]: Add an empty line here

File: src/views/users/createUserForm.hbs [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+                    <input type="text" class="form-control" name="pwd" placeholder="Contraseña" autofocus/>
+                </div>
+
+                <div class="form-group">
+                    <button class="btn btn-primary btn-block" type="submit">Guardar</button>
+                </div>
+            </form>
+        </div>
+    </div>
+</div>
```
Comment (pinto1997) [inline src/views/users/createUserForm.hbs:39]: Add an empty line here

File: src/services/user.service.js [hunk 16]
```diff
@@ -1,13 +1,14 @@
 const axios = require("axios");
-const User = require("../models/User");
-const USERS_SERVICE_URI = "https://at-sso-api.herokuapp.com/api/v1/user";
+const AT_SSO_SERVICE_URI = "https://at-sso-api.herokuapp.com/api/v1/user";
```
Comment (at-internship) [inline src/services/user.service.js]: const AT_SSO_SERVICE_URI = process.env.AT_SSO_SERVICE_URI;

**Review by at-internship — CHANGES_REQUESTED** [review]
It's almost done, just take a look at how you get const AT_SSO_SERVICE_URI

**Review by at-internship — APPROVED** [review]
Go go go

**Review by pinto1997 — APPROVED** [review]

**Pull request merged**